pkg/schema/type.go

```diff
@@ -107,6 +107,8 @@ func (a ArrayItemType) String() string {
 }
 func (m ScalarType) String() string {
 	switch m.Value.(type) {
+	case float64:
+		return "number"
```

Comment:
any reason why not use `float`?

Reply:
Does go have a `float` type? I thought it only included float32 and float64. I'll check again :)

Reply:
im talking about the string `"number"`, should probably be changed to `"float"` because ytt users do not know any concept called number.

Reply:
The objective with that wording was to make sure people would know that ints would be valid as well. 

Made a quick slack poll to see how folx feel:
https://kubernetes.slack.com/archives/CH8KCCKA5/p1618507762314100

Reply:
i dont think there is major language out there that would not accept int value into a float type. e.g. in Go `var x float64 = 4` is totally valid.

Reply:
I see the motivation to get to the most precise term and I appreciate that effort. Getting to precision is an important factor... and it needs to be contextualized in order to avoid a local optimization ...

So, stepping back, thinking of the overall user experience, if YAML has a type called "float" and Starlark has a type called "float", it is intuitive to call such a value a "float" here.